File: Aplicación Escritorio/Modelo/Ruta.cs
using System;
using System.IO;
using System.Reflection.Metadata;
using System.Text.Json.Serialization;
using System.Text.RegularExpressions;
using System.Xml.Linq;

namespace Modelo
{
    public class Ruta
    {
        public int idRuta { get; set; }
        public String nombre { get; set; }
        public String nombreInicioruta { get; set; }

        public String nombreFinalruta { get; set; }

        public Double latitudInicial { get; set; }

        public Double latitudFinal { get; set; }

        public Double longitudInicial { get; set; }

        public Double longitudFinal { get; set; }

        public Double distancia { get; set; }

        public TimeOnly duracion { get; set; }

        public int desnivelPositivo { get; set; }


        public int desnivelNegativo { get; set; }

        public int desnivelAcumulado { get; set; }

        public Double altitudMax { get; set; }

        public Double altitudMin { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public CLASIFICACION clasificacion { get; set; }

        public Byte nivelEsfuerzo { get; set; }

        public Byte nivelRiesgo { get; set; }

        public Boolean estadoRuta { get; set; }

        public Byte tipoTerreno { get; set; }

        public Byte indicaciones { get; set; }

        public String temporadas { get; set; }

        public Boolean accesibilidad { get; set; }

        public Boolean rutaFamiliar { get; set; }


        public String archivoGPX { get; set; }

        public String recomendacionesEquipo { get; set; }

        public String zonaGeografica { get; set; }

        public Double mediaEstrellas { get; set; }

        public Usuario usuarioIdusuario { get; set; }
        public List<PuntoRuta> puntos { get; set; }

        public List<Valoracion> valoraciones { get; set; }

        public List<Resena> resenas { get; set; }


    }
}

File: AplicacionMovil/app/src/main/java/com/example/retodam2rutas/entities/Ruta.kt
package com.example.retodam2rutas.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.time.LocalTime

@Entity(tableName = "rutas",
    foreignKeys = [
        ForeignKey(
            entity = Usuario::class,
            parentColumns = ["idUsuario"],
            childColumns = ["usuarioId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["usuarioId"])]
)
data class Ruta(

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "idRuta")
    val id: Int,

    @ColumnInfo(name = "nombre")
    val nombre: String,

    @ColumnInfo(name = "nombre_inicioruta", defaultValue = "Inicio")
    val nombreInicioruta: String = "Inicio",

    @ColumnInfo(name = "nombre_finalruta",defaultValue = "Fin")
    val nombreFinalruta: String = "Fin",

    @ColumnInfo(name = "latitudInicial")
    val latitudInicial: Double,

    @ColumnInfo(name = "latitudFinal")
    val latitudFinal: Double,

    @ColumnInfo(name = "longitudInicial")
    val longitudInicial: Double,

    @ColumnInfo(name = "longitudFinal")
    val longitudFinal: Double,

    @ColumnInfo(name = "distancia")
    var distancia: Double,

    @ColumnInfo(name = "duracion")
    var duracion: LocalTime,

    @ColumnInfo(name = "desnivelPositivo")
    var desnivelPositivo: Int,

    @ColumnInfo(name = "desnivelNegativo")
    var desnivelNegativo: Int,

    @ColumnInfo(name = "desnivelAcumulado")
    var desnivelAcumulado: Int,

    @ColumnInfo(name = "altitudMax")
    var altitudMax: Double,

    @ColumnInfo(name = "altitudMin")
    var altitudMin: Double,

    @ColumnInfo(name = "clasificacion", defaultValue = "LINEAL")
    var clasificacion: CLASIFICACION = CLASIFICACION.LINEAL,

    @ColumnInfo(name = "nivelEsfuerzo")
    var nivelEsfuerzo: Byte? = null,

    @ColumnInfo(name = "nivelRiesgo")
    var nivelRiesgo: Byte? = null,

    @ColumnInfo(name = "estadoRuta", defaultValue = "0")
    var estadoRuta: Boolean = false,

    @ColumnInfo(name = "tipoTerreno")
    val tipoTerreno: Byte? = null,

    @ColumnInfo(name = "indicaciones")
    val indicaciones: Byte? = null,

    @ColumnInfo(name = "temporadas")
    val temporadas: String? = null,

    @ColumnInfo(name = "accesibilidad")
    val accesibilidad: Boolean = false,

    @ColumnInfo(name = "rutaFamiliar")
    val rutaFamiliar: Boolean = false,

    @ColumnInfo(name = "archivoGPX")
    val archivoGPX: String? = null,

    @ColumnInfo(name = "recomendacionesEquipo")
    val recomendacionesEquipo: String? = null,

    @ColumnInfo(name = "zonaGeografica")
    val zonaGeografica: String? = null,

    @ColumnInfo(name = "mediaEstrellas")
    val mediaEstrellas: Double? = null,

    @ColumnInfo(name = "usuarioId")
    val usuarioId: Int
)